src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -785,7 +790,9 @@ export function findPreferredKernel(
                 subScore === 5 && // This is a bit flakey. Number isn't consistent. Should probably just make the order of kernelspecs have the preferred one first
                 score === 5 &&
                 (metadata.kind === 'startUsingPythonInterpreter' ||
-                    (metadata.kind === 'startUsingKernelSpec' && metadata.kernelSpec.language === PYTHON_LANGUAGE)) &&
+                    metadata.kind === 'startUsingRemoteKernelSpec' ||
```

Comment:
Why would the logic be different for remote non python kernel specs?

Reply:
Not sure, this is exisit code, let me see.